pkgs/tools/security/hiprompt-gtk-py/default.nix

```
@@ -0,0 +1,54 @@
+{ lib
+, stdenv
+, fetchFromSourcehut
+, desktop-file-utils
+, glib
+, gobject-introspection
+, gtk3
+, gtk-layer-shell
+, meson
+, ninja
+, pkg-config
+, python3
+, wrapGAppsHook
+}:
+
+stdenv.mkDerivation rec {
+  pname = "hiprompt-gtk-py";
+  version = "unstable-2022-06-17";
+
+  src = fetchFromSourcehut {
+    name = pname + "-src";
+    owner = "~sircmpwn";
+    repo = pname;
+    rev = "f74499302bdd6558d4644c25e15c9b5c482270ea";
+    hash = "sha256-/vYg7q4ZVakJ0BeULxJWXnBYb75tkgVVXbHYwnN6dZ4=";
+  };
+
+  nativeBuildInputs = [
+    desktop-file-utils
+    glib
+    meson
+    ninja
+    pkg-config
+    wrapGAppsHook
+  ];
+
+  buildInputs = [
+    glib
+    gobject-introspection
+    gtk3
+    gtk-layer-shell
+    (python3.withPackages (pp: with pp; [
+      pygobject3
+    ]))
+  ];
+
+  meta = with lib; {
```

```suggestion
  meta = with lib; {
    mainProgram = "hiprompt-gtk";
```

so that we can use `lib.getExe` in https://github.com/nix-community/home-manager/pull/4655